app/controllers/api/v1/events_controller.rb

```diff
@@ -94,15 +97,16 @@ def create_params
 
       def batch_params
         params
-          .require(:event)
           .permit(
-            :transaction_id,
-            :external_customer_id,
-            :code,
-            :timestamp,
-            external_subscription_ids: [],
-            properties: {},
-          )
+            events: [
+              :transaction_id,
+              :external_customer_id,
+              :code,
+              :timestamp,
+              :external_subscription_id,
+              properties: {},
```

Comment:
Style/HashAsLastArrayItem: Wrap hash in `{` and `}`. (https://rubystyle.guide#hash-literal-as-last-array-item)